### PR TITLE
Update embedded JDKs to latest patch release

### PR DESCRIPTION
**Pull request by fmeum**
Also switch back to Zulu where possible.

### CODE REVIEW

**Comment by fmeum** [comment]
Mirror request: https://github.com/bazelbuild/bazel/issues/23914

**Comment by fmeum** [comment]
FYI @sluongng This may fix the macOS JDK crashes you have seen.

**Comment by sluongng** [comment]
Would this make it to 7.4 and 8.0 in time? 🤔

**Comment by fmeum** [comment]
@bazel-io fork 7.4.0

**Comment by fmeum** [comment]
@bazel-io fork 8.0.0

File: repositories.bzl [hunk 1]
```diff
@@ -74,54 +74,49 @@ def embedded_jdk_repositories():
...
     )
-
-    # JDK21 unavailable from zulu, we'll use Microsoft's OpenJDK build instead.
     http_file(
         name = "openjdk_win_arm64_vanilla",
-        sha256 = "975603e684f2ec5a525b3b5336d6aa0b09b5b7d2d0d9e271bd6a9892ad550181",
+        integrity = "sha256-n4c+zPAwsdPch57B6w/14Rv3YALcgcXGRMNGK/bFFGs=",
         downloaded_file_path = "zulu-win-arm64.zip",
-        url = "https://aka.ms/download-jdk/microsoft-jdk-21.0.0-windows-aarch64.zip",
+        url = "https://cdn.azul.com/zulu/bin/zulu21.36.17-ca-jdk21.0.4-win_aarch64.zip",
```
Comment (meteorcloudy) [inline repositories.bzl:121]: Thanks, can you also remove 
https://github.com/bazelbuild/bazel/blob/7c0ddd2ecf95ae79afbc36247f403d717c32ac97/bazel_downloader.cfg#L7-L8

**Review by meteorcloudy — APPROVED** [review]

File: repositories.bzl (hunk 1 above)
Comment (fmeum) [reply repositories.bzl:121]: Done!

**Review by meteorcloudy — APPROVED** [review]

**Comment by iancha1992** [comment]
The changes in this PR have been included in [Bazel 7.4.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.4.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting `USE_BAZEL_VERSION=7.4.0rc1`. Thanks!

**Pull request closed without merging**